src/tlo/methods/rti.py

```diff
@@ -1525,6 +1530,107 @@ def initialise_simulation(self, sim):
         sim.schedule_event(RTI_Check_Death_No_Med(self), sim.date + DateOffset(months=0))
         # Begin logging the RTI events
         sim.schedule_event(RTI_Logging_Event(self), sim.date + DateOffset(months=1))
+        # Look-up consumable item codes
```

Comment:
As discussed on slack, you're 100% right about this and it's a good spot. Currently the HSI does the (expensive) lookup of the item codes every time it's called, when it would be more efficient (and more readable) if the consumables are declared once, and all in one go, and the resulting item_codes saved on the module. The way you've set this up in the commented-out section looks right, but I would recommend we put it in a separate PR. So, please could you roll-it back on this branch (so this PR is focussed only on the change we want to make about HSI recurrence), but open another PR that makes that change?